pkg/network/node/egressip.go

```diff
@@ -85,34 +87,49 @@ func (eip *egressIPWatcher) Start(osdnInformers osdninformers.SharedInformerFact
 	return nil
 }
 
-func (eip *egressIPWatcher) Synced() {
+func (eip *egressIPWatcher) SyncIPAssignments() error {
 	link, _, err := GetLinkDetails(eip.localIP)
 	if err != nil {
-		// shouldn't happen, but obviously there's nothing to clean up...
-		return
+		return fmt.Errorf("could not get %s link details %v", eip.localIP, err)
 	}
 	label, err := egressIPLabel(link)
 	if err != nil {
-		klog.Errorf("Could not check for stale egress IPs: %v", err)
-		return
+		return fmt.Errorf("invalid egress IP label: %v", err)
 	}
 	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
 	if err != nil {
-		klog.Errorf("Could not check for stale egress IPs: %v", err)
-		return
+		return fmt.Errorf("could not check for egress IPs: %v", err)
 	}
 
+	existingIPs := sets.NewString()
 	for _, addr := range addrs {
 		ip := addr.IP.String()
-		if addr.Label == label && eip.iptablesMark[ip] == "" {
-			klog.Infof("Cleaning up stale egress IP %s", addr.IP.String())
-			err = netlink.AddrDel(link, &addr)
-			if err != nil {
-				klog.Errorf("Could not clean up stale egress IP: %v", err)
+		if addr.Label == label {
+			if eip.iptablesMark[ip] == "" {
+				klog.Infof("Cleaning up stale egress IP %s", addr.IP.String())
+				err = netlink.AddrDel(link, &addr)
+				if err != nil {
+					klog.Errorf("Could not clean up stale egress IP: %v", err)
+				}
+			} else {
+				existingIPs.Insert(ip)
 			}
 		}
 	}
+	for ip, mark := range eip.iptablesMark {
+		if !existingIPs.Has(ip) {
+			klog.Infof("Assign missing egress IP %s", ip)
```

Comment:
```suggestion
			klog.Infof("Assigned missing egress IP %s", ip)
```

Reply:
changed to `klog.Infof("Assigning missing egress IP %s", ip)` to match: https://github.com/openshift/sdn/blob/76f2671ee199483a86df10471918a6bf7f857b4a/pkg/network/node/egressip.go#L109